File: src/main/java/com/ggardet/doctor/spi/dto/SpecialtyDTO.java
package com.ggardet.doctor.spi.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

import java.util.UUID;

public record SpecialtyDTO(
        UUID id,

        @NotBlank String name,

        String description,

        @Positive Integer requiredYearsTraining
) {
}

File: src/main/kotlin/com/ggardet/doctor/spi/dto/SpecialtyDTO.kt
package com.ggardet.doctor.spi.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import java.util.*

data class SpecialtyDTO(
    val id: UUID?,
    
    @field:NotBlank
    val name: String,
    
    val description: String?,
    
    @field:Positive
    val requiredYearsTraining: Int?
)